docs/src/tut/ticket-sales/index.md

```diff
@@ -0,0 +1,274 @@
+
+Let's move back to the Reach file and actually create our API function.
+
+This program ends up being short and compact because of the `{!rsh} parallelReduce` control structure that we'll implement to allow access to our API functions in a looping construct.
+
+Read all about it in our @{seclink("guide-parallelReduce")}.
+
+In this case, we want the loop to allow the function to be callable until the contract is out of tokens. This is an important piece of information for our design.
+```
+load: /examples/ticket-sales/index.rsh
+md5: c425745032273893d106fe3de005f15e
+range: 25-28
+```
+- Line 25 declares a new `{!rsh} parallelReduce` and sets it up to track a single value `ticketsSold`, which is initialized to zero.
+- Line 26 states an `{!rsh} invariant` about our loop that the network token balance will always equal `cost * ticketsSold`.
+- Line 27 states an `{!rsh} invariant` about our loop that the non-network token balance will always equal the initial `supply - ticketsSold`.
+- Line 28 sets our `{!rsh} while` loop to run until `ticketsSold == supply` at which point it will exit the `{!rsh} parallelReduce`.
+
+It is important to note the relationship between the values above. 
+
+The Reach compiler will check your DApp for *all* of the possibilities related to your program values in an *unbounded* way, bound only by the limits of the data type you are working with. 
+
+Reach will put bounds on those checks as determined by your verification checks and `{!rsh} invariant`s.
+
+By tightly tracking all of the `{!rsh} invariant` values in our `{!rsh} parallelReduce` LHS and `{!rsh} while` loop -- and relating them to the contract balance of network tokens and non-network tokens, we can satisfy the Reach Verification Engine to produce no errors.
+
+Read more about @{seclink("guide-assert")}.
+
+Understanding this relationship is key to building Reach DApps.
+
+Now we can actually define our API function.
+```
+load: /examples/ticket-sales/index.rsh
+md5: c425745032273893d106fe3de005f15e
+range: 29-36
+```
+- Line 29 defines an API macro `{!rsh} .api_` from the `B` API with the name `buyTicket`. It takes no parameters.
+- Line 30 is an extra verification check. Removing this from the program *will not* introduce compiler errors, but is good for defensive programming. (note to Jay: the only purpose this check serves is to negate split second API calls from being made before the loop exits properly. Is this even possible, or should we just remove this?)
+- Line 31 starts the outer `{!rsh} return`. Here we sepecify a `PAY_EXPR` of `cost` and declare our `{!rsh} return` function `ret`.
+- Line 32 sends one non-network token (ticket) to the API caller.
+- Line 33 returns `{!rsh} null` to the caller.
+- Line 34 is our inner `{!rsh} return`, where we increment `{!rsh} parallelReduce` LHS value `ticketsSold`.
+
+:::note
+On incrementing, Reach *does not* support incrementing with `i++`. 
```

Comment:
, because that is mutation and Reach insists that we always return new, updated values.